addons/hr_payroll/wizard/payslip_lines_contribution_register.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import time
+from datetime import datetime
+from dateutil import relativedelta
+from odoo import api, fields, models
+
+
+class PayslipLinesContributionRegister(models.TransientModel):
+    _name = 'payslip.lines.contribution.register'
+    _description = 'PaySlip Lines by Contribution Registers'
+
+    date_from = fields.Date(required=True, default=lambda *a: time.strftime('%Y-%m-01'))
+    date_to = fields.Date(required=True, default=lambda *a: str(datetime.now() + relativedelta.relativedelta(months=+1, day=1, days=-1))[:10])
+
+    @api.multi
+    def print_report(self):
+        datas = {
+             'ids': self.env.context.get('active_ids', []),
```

Comment:
is it required to pass [] in self.env.context ? no need of that.. 
self.env.context.get('active_ids') enought